katas/content/phase_estimation/index.md

```diff
@@ -0,0 +1,124 @@
+
+$$|\lambda| = 1$$
+
+Thus, they can be written in the following form:
+$$\lambda = e^{i\theta}$$
+
+The value $\theta$ is called an *eigenphase* that corresponds to this eigenvector.
+
+> How can you prove that the modulus of an eigenvalue of a unitary matrix equals $1$?
+> 
+> On one hand, by definition of an eigenvalue, 
+> $$U \ket{v} = \lambda \ket{v}$$
+> $$|U \ket{v}| = |\lambda| \cdot |\ket{v}|$$
+> On the other hand, using the properties of the unitary matrix, you can write the following equation:
+> $$|U \ket{v}|^2 = \bra{v} U^\dagger U \ket{v} = \bra{v} U^{-1} U \ket{v} = \bra{v} I \ket{v} = \braket{v|v} = |\ket{v}|^2$$
+>
+> From these two equations, you get the following equality:
+> $$(|\lambda| \cdot |\ket{v}|)^2 = |\ket{v}|^2$$
+> And then, finally:
+> $$|\lambda| = 1$$
+
+If the quantum gate is self-adjoint, that is, its matrix equals its inverse $U^{-1} = U$, the eigenvalues of this matrix can only be $+1$ and $-1$, with eigenphases $0$ and $\pi$, respectively.
+
+> You can prove this in a similar manner, using the defintion of an eigenvalue:
+> $$U^2 \ket{v} = U(U \ket{v}) = U(\lambda \ket{v}) = \lambda U \ket{v} = \lambda^2 \ket{v}$$
+> At the same time,
+> $$U^2 \ket{v} = UU \ket{v} = U^(-1)U \ket{v} = I \ket{v} = \ket{v}$$
+> So you can conclude that $\lambda^2 = 1$.
+
+For example, the $Z$ gate has two eigenvctors:
+- $\ket{0}$, with eigenvalue $1$
+- $\ket{1}$, with eigenvalue $-1$
+
+
+@[exercise]({
+    "id": "phase_estimation__eigenvalues_s",
+    "title": "Find Eigenvalues of the S Gate",
+    "path": "./eigenvalues_s/"
+})
+
+@[exercise]({
+    "id": "phase_estimation__eigenvectors_x",
+    "title": "Find Eigenvectors of the X Gate",
+    "path": "./eigenvectors_x/"
+})
+
+@[exercise]({
+    "id": "phase_estimation__state_eigenvector",
+    "title": "Is Given State an Eigenvector of the Gate?",
+    "path": "./state_eigenvector/"
+})
+
+
+@[section]({
+    "id": "phase_estimation__problem",
+    "title": "Phase Estimation Problem"
+})
+
+The phase estimation problem is formulated as follows. 
+
+You are given a unitary operator $U$ and its eigenvector $\ket{\psi}$. The eigenvector is given as a way to prepare it,
+for example, a unitary $P$ that, when applied to the $\ket{0}$ state, converts it to the state $\ket{\psi}$.
+
+Your goal is to find the eigenvalue $\lambda$ associated with this eigenvector, or, in a more common formulation, the corresponding eigenphase $\theta$:
+
+$$U\ket{\psi} = e^{2 \pi i \theta} \ket{\psi}, \theta = ?$$
+
+The value of $\theta$ is defined to be between $0$ and $1$, since any value outside of this range has an equivalent value within it. Instead of discussing the floating-point number $\theta$, sometimes its representation as a binary fraction with $n$ digits is used:
```

Comment:
nit:
```suggestion
The value of $\theta$ is defined to be between $0$ and $1$, since any value outside of this range has an equivalent value within it. Instead of representing $\theta$ as a decimal, sometimes it is represented as a binary fraction with $n$ digits:
```